std/algorithm/searching.d

```diff
@@ -2900,7 +2921,27 @@ true if the first element matches the given element according to the given
 predicate, and the range has been advanced by one element; otherwise false, and
 the range is left untouched.
  */
-bool skipOver(alias pred = "a == b", R, E)(ref R r, E e)
+bool skipOver(R, E)(ref R r, E e)
+    if (isInputRange!R && is(typeof(r.front == e) : bool))
+{
+    static if (isSomeString!R && isSomeChar!E &&
+        r[0].sizeof == e.sizeof)
+    {
+        if (r.empty || r[0] != e)
```

Comment:
This eats into unicode characters, obviously. Consider `string s = "祝"; skipOver(s, '\347');`. The string version has its merits, since both strings are probably valid. The char version seems unwarranted, on second thought. I'd drop it.